src/components/activity-client-interface.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsContext} from '../proto/api_messages';
+import {ActivityIframePort} from 'web-activities/activity-ports';
+import {ActivityMode, ActivityResult} from 'web-activities/activities';
+
+/**
+ * @typedef {!Array<(string|boolean|number|null|!Array<(string|boolean|number|null)>)>} SerializedBuffer
```

Comment:
Is this just a bad copy-paste? Speaking of which - I still think we should move all of our protolite `!Array<(string|.......)>` to just `!Array`. Such a big union of types is somewhat confusing. It's also incomplete.